Car_owners/tests/test_views.py

```diff
@@ -2,268 +2,252 @@
 import pytest
 from rest_framework import status
 from rest_framework.test import APIClient
+from application.models import Owner, Car
 from application.views import OwnerViewSet, CarViewSet
 
 
-@pytest.fixture
-def api_client() -> APIClient:
-    yield APIClient()
-
-
-@pytest.fixture
-def owner_data() -> dict[str, str]:
-    owner_data = {"name": "Andrzej", "surname": "Starczyk", "phone": "575849675"}
-    return owner_data
-
-
-@pytest.fixture
-def car_data(api_client: APIClient, owner_data: dict[str, str]) -> dict[str, str | int]:
-    response_create_owner = api_client.post(
-        "/app/owners/", data=owner_data, format="json"
-    )
-    owner_id = response_create_owner.data["id"]
-    car_data = {
-        "brand": "Ford",
-        "model": "Mondeo",
-        "production_date": "2023-01-01",
-        "owner": owner_id,
-    }
-
-    return car_data
-
-
-@pytest.mark.django_db
-def test_create_and_get_owner(
-    api_client: APIClient, owner_data: dict[str, str]
-) -> None:
-    # create owner
-    response_create_owner = api_client.post(
-        "/app/owners/", data=owner_data, format="json"
-    )
-    owner_id = response_create_owner.data["id"]
-    assert response_create_owner.status_code == status.HTTP_201_CREATED
-    assert all(
+class TestsOwnerViews:
+    @pytest.mark.django_db
+    def test_create_owner(
+        self, api_client: APIClient, valid_owner_data: dict[str, str]
+    ) -> None:
+        # create owner
+        response_create_owner = api_client.post(
+            "/app/owners/", data=valid_owner_data, format="json"
+        )
+        assert response_create_owner.status_code == status.HTTP_201_CREATED
+        assert all(
+            [
+                response_create_owner.data[key] == valid_owner_data[key]
+                for key in valid_owner_data.keys()
+            ]
+        )
+
+    @pytest.mark.django_db
+    def test_get_owner(
+        self, api_client: APIClient, valid_owner_model_data: Owner
+    ) -> None:
+        owner_id = valid_owner_model_data.id
+        # get owner
+        response_get_owner = api_client.get(f"/app/owners/{owner_id}/", format="json")
+        valid_owner_model_data_dict = vars(valid_owner_model_data)
```

Comment:
that's sketchy, better to just use serializer you already have

Reply:
and then instead of comparing just keys or just Andrzej you can compare json vs json